src/code.cloudfoundry.org/cni-wrapper-plugin/netrules/rule_converter.go

```diff
@@ -59,6 +59,22 @@ func (c *RuleConverter) BulkConvert(ruleSpec []Rule, logChainName string, global
 	return iptablesRules
 }
 
+func (c *RuleConverter) DeduplicateRules(iptablesRules []rules.IPTablesRule) []rules.IPTablesRule {
+	keys := make(map[string]bool)
+	deduped_rules := []rules.IPTablesRule{}
```

Comment:
Aren't we creating a copy of the rules set here? What would be the memory impact, we have seen cases with millions of rules?

Reply:
We create a map, in which we use the rules as keys and the value is true or false. If on the current iteration of the rules list we have a rule which is not part of the keys map, we add it to keys and set the value to true. This way if a duplicate of that rule is present, we will skip adding it to the dedupedRules, since the value of that  rule is set to true. As to the memory concern, this way of removing duplicates was the one I saw used the most in go. Honestly I do not know how big will this impact the performance. Open to suggestions here